primitives/src/lib.rs

```diff
@@ -696,7 +696,10 @@ impl StaticLookup for BalanceConversion {
 	}
 
 	fn unlookup(stellar_stroops: Self::Target) -> Self::Source {
-		(stellar_stroops * CONVERSION_RATE as i64) as u128
+		let conversion_rate = i64::try_from(CONVERSION_RATE).unwrap_or(i64::MAX);
+
+		let value = stellar_stroops * conversion_rate;
```

Comment:
Maybe we should even use `saturating_mul()` here, just to be safe

Reply:
@b-yap WDYT?

Reply:
done